src/EnergyPlus/MixedAir.cc

```diff
@@ -791,6 +791,40 @@ void SimOAController(EnergyPlusData &state, std::string const &CtrlName, int &Ct
         state.dataMixedAir->GetOAControllerInputFlag = false;
     }
 
+    // check that the economizer staging operation EconomizerFirst is only used with an sensible load-based controlled AirLoopHVAC:UnitarySystem
+    if (AirLoopNum > 0) {
+        auto &primaryAirSystems = state.dataAirSystemsData->PrimaryAirSystems(AirLoopNum);
+        bool sensLoadCtrlUnitarySystemFound = false;
+        if (primaryAirSystems.EconomizerStagingCheckFlag == false) {
+            OAControllerNum = UtilityRoutines::FindItemInList(CtrlName, state.dataMixedAir->OAController);
+            if (state.dataMixedAir->OAController(OAControllerNum).EconomizerStagingOperation == EconomizerFirst) {
+                for (int BranchNum = 1; BranchNum <= primaryAirSystems.NumBranches; ++BranchNum) {
+                    for (int CompNum = 1; CompNum <= primaryAirSystems.Branch(BranchNum).TotalComponents; ++CompNum) {
+                        if (primaryAirSystems.Branch(BranchNum).Comp(CompNum).CompType_Num == SimAirServingZones::CompType::UnitarySystemModel) {
+                            std::string unitarySystemName = primaryAirSystems.Branch(BranchNum).Comp(CompNum).Name;
+                            int unitarySystemNum = UtilityRoutines::FindItemInList(
+                                unitarySystemName, state.dataUnitarySystems->unitarySys, state.dataUnitarySystems->numUnitarySystems);
+                            if (state.dataUnitarySystems->unitarySys[unitarySystemNum - 1].m_ControlType ==
+                                UnitarySystems::UnitarySys::UnitarySysCtrlType::Load) {
```

Comment:
Not part of this PR, but I would like to point out that in this fully-resolved name, `UnitarySys` appears THREE times! So, what you're telling me is that this is related to unitary systems?